### DISCUSSION TITLE
5.5.0-1

### DISCUSSION THREAD

**Discussion by electronstudio** [electronstudio/jaylib#67] [Announcements]
* fix missing newRectangle method

<hr /><em>This discussion was created from the release <a href='https://github.com/electronstudio/jaylib/releases/tag/v5.5.0-1'>5.5.0-1</a>.</em>

**Comment by Luigi654** [comment]
can add 32 bit support on all OS?, i have a potato pc using java 8

**Reply by electronstudio** [reply]
It should build on 32 bit if you build it yourself. 

The problem is GitHub Actions doesn’t have 32 bit machines for me to build it on. So someone would have to figure out how to cross compile 32 bit on 64 bit machine.

**Reply by electronstudio** [reply]
does a potato PC support opengl 3.3?

**Reply by Luigi654** [reply]
> It should build on 32 bit if you build it yourself.
> 
> The problem is GitHub Actions doesn’t have 32 bit machines for me to build it on. So someone would have to figure out how to cross compile 32 bit on 64 bit machine.

in this version https://github.com/CreedVI/Raylib-J/releases/tag/v0.5 have 32 bit support

**Reply by Luigi654** [reply]
yes on Intel HD graphics, sadly i havent nothing to update my pc from year 2020, here the economy is very bad, i knows some users who still 32 bits

**Reply by electronstudio** [reply]
Raylib-J is completely different thing, a pure Java library, that just uses the Raylib name - it's not a binding to Raylib - so that doesn't help.

But I think I have got a build working.  I'm just not sure if it's better to target OpenGL 2.0 because most people with potato PCs wont have modern graphics cards either.

**Reply by electronstudio** [reply]
does this work?  [jaylib55-win32-open20-test1.zip](https://github.com/user-attachments/files/17969101/jaylib55-win32-open20-test1.zip)

**Reply by Luigi654** [reply]
> does this work? [jaylib55-win32-open20-test1.zip](https://github.com/user-attachments/files/17969101/jaylib55-win32-open20-test1.zip)

works perfectly

run:
INFO: Initializing raylib 5.5
INFO: Platform backend: DESKTOP (GLFW)
INFO: Supported raylib modules:
INFO:     > rcore:..... loaded (mandatory)
INFO:     > rlgl:...... loaded (mandatory)
INFO:     > rshapes:... loaded (optional)
INFO:     > rtextures:. loaded (optional)
INFO:     > rtext:..... loaded (optional)
INFO:     > rmodels:... loaded (optional)
INFO:     > raudio:.... loaded (optional)
INFO: DISPLAY: Device initialized successfully
INFO:     > Display size: 1366 x 768
INFO:     > Screen size:  800 x 450
INFO:     > Render size:  800 x 450
INFO:     > Viewport offsets: 0, 0
INFO: GLAD: OpenGL extensions loaded successfully
INFO: GL: Supported extensions count: 221
INFO: GL: OpenGL device information:
INFO:     > Vendor:   Intel
INFO:     > Renderer: Intel(R) HD Graphics 400
INFO:     > Version:  4.4.0 - Build 20.19.15.5171
INFO:     > GLSL:     4.40 - Build 20.19.15.5171
INFO: GL: VAO extension detected, VAO functions loaded successfully
INFO: GL: NPOT textures extension detected, full NPOT textures supported
INFO: GL: DXT compressed textures supported
INFO: GL: ETC2/EAC compressed textures supported
INFO: PLATFORM: DESKTOP (GLFW - Win32): Initialized successfully
INFO: TEXTURE: [ID 1] Texture loaded successfully (1x1 | R8G8B8A8 | 1 mipmaps)
INFO: TEXTURE: [ID 1] Default texture loaded successfully
INFO: SHADER: [ID 1] Vertex shader compiled successfully
INFO: SHADER: [ID 2] Fragment shader compiled successfully
INFO: SHADER: [ID 3] Program shader loaded successfully
INFO: SHADER: [ID 3] Default shader loaded successfully
INFO: RLGL: Render batch vertex buffers loaded successfully in RAM (CPU)
INFO: RLGL: Render batch vertex buffers loaded successfully in VRAM (GPU)
INFO: RLGL: Default OpenGL state initialized successfully
INFO: TEXTURE: [ID 2] Texture loaded successfully (128x128 | GRAY_ALPHA | 1 mipmaps)
INFO: FONT: Default font loaded successfully (224 glyphs)
INFO: SYSTEM: Working Directory: D:\Documents\NetBeansProjects\open_smash_source_3d
INFO: TIMER: Target time per frame: 16.667 milliseconds
INFO: TEXTURE: [ID 2] Unloaded texture data from VRAM (GPU)
INFO: SHADER: [ID 3] Default shader unloaded successfully
INFO: TEXTURE: [ID 1] Default texture unloaded successfully
INFO: Window closed successfully
BUILD SUCCESSFUL (total time: 26 seconds)

![imagen](https://github.com/user-attachments/assets/31409155-370a-4f8f-af04-383ed5138d11)

**Reply by electronstudio** [reply]
great i'll add it to the next release

**Reply by Luigi654** [reply]
> great i'll add it to the next release

thanks :3, include others OS too?

**Reply by electronstudio** [reply]
You mean Linux?  That's more complicated because the Linux build is done on Ubuntu 20.04.  There is no 32 bit version of Ubuntu 20.04, so what version would we expect the user of the library to be running?

**Reply by Luigi654** [reply]
> You mean Linux? That's more complicated because the Linux build is done on Ubuntu 20.04. There is no 32 bit version of Ubuntu 20.04, so what version would we expect the user of the library to be running?

In all possible as you can, like as raylib-j of creedVI have support all OS 32 bits, i think is same as 64 bit but less power

**Reply by electronstudio** [reply]
As I already said, comparisons to Raylib-J are not relevant because that is a Java library, not a binding for a C library.

I could setup a cross compiler on Ubuntu 20.04 to compile for x86, but Linux isn't backwards compatible like Windows is.  If the user of the library was running Ubuntu 18 (the last version that supported x86) then the binary I compiled wouldn't work.

So let's say we do setup an x86 Ubuntu 18 build environment (not supported on Github Actions).  Is our end user going to be running Ubuntu 18?  We're talking about someone with a CPU that hasn't been manufactured for 20 years, so I would think their version of Linux is going to be much, much older.

**Reply by Luigi654** [reply]
Ok thanks for the info, also, the test you gives me works with shaders of lighting

**Reply by electronstudio** [reply]
Looking at what Raylib does with their official Linux x86 releases, they compile on Ubuntu 20.04 with cross compiler for x86, but they only ship a static lib, not a shared one, so maybe they had some issue with it.  So I don't know if doing that would be useful for anyone or not.  I'm assuming you only need Windows?  So maybe we wait until somebody actually needs x86 Linux and asks for it.

If you need OpenGL 3.3 for shaders let me know, current build is 2.1 in order to be more compatible.

**Reply by Luigi654** [reply]
Me only who can handle 32 bits, windows works perfect with the test you gives me, but if i want produce the program, i need porting for others OS

**Reply by electronstudio** [reply]
Well specifically what Linux distro do you plan to run on?

**Reply by Luigi654** [reply]
The most used that, ubuntu, no?

**Reply by electronstudio** [reply]
Ubuntu hasn't supported 32 bit since 2018, so which version of Ubuntu are you planning to run?

**Reply by Luigi654** [reply]
> Ubuntu hasn't supported 32 bit since 2018, so which version of Ubuntu are you planning to run?

Ubuntu 24.04.1 LTS

**Reply by electronstudio** [reply]
You can't, that is 64 bit only.

**Reply by Luigi654** [reply]
> You can't, that is 64 bit only.

Ah, ok

**Reply by electronstudio** [reply]
I tried a cross compile on Ubuntu 20.04 but it didn’t work because they don’t even ship the 32 bit x11 lib needed anymore. So I suggest if you do install a 32 bit Linux you compile Jaylib yourself - it should be very easy to compile 32 bit Jaylib on 32 bit Linux on 32 bit machine. The difficulty is I don’t have that.

**Reply by Luigi654** [reply]
> I tried a cross compile on Ubuntu 20.04 but it didn’t work because they don’t even ship the 32 bit x11 lib needed anymore. So I suggest if you do install a 32 bit Linux you compile Jaylib yourself - it should be very easy to compile 32 bit Jaylib on 32 bit Linux on 32 bit machine. The difficulty is I don’t have that.

Ok thanks, you will add only 32 bit support for windows?

**Reply by electronstudio** [reply]
I feel like we are going around and around in circles.

**Reply by electronstudio** [reply]
anyway, released https://github.com/electronstudio/jaylib/releases/tag/v5.5.0-2
